src/azsdkdemoprocessor/src/main/java/demo/sdk/az/App.java

```diff
@@ -0,0 +1,93 @@
+package demo.sdk.az;
+
+import ch.qos.logback.classic.Logger;
+import com.azure.identity.*;
+import com.azure.storage.queue.*;
+import com.microsoft.applicationinsights.logback.ApplicationInsightsAppender;
+import io.github.cdimascio.dotenv.*;
+import java.util.*;
+import org.slf4j.LoggerFactory;
+
+public class App {
+  static final Logger logger = (Logger) LoggerFactory.getLogger("root");
+  static final Dotenv env = Dotenv
+    .configure()
+    .ignoreIfMissing()
+    .systemProperties()
+    .directory("../../")
+    .load();
+
+  public static void main(String[] args) {
+    try {
+      App app = new App();
+      app.setInstrumentationKey();
+      app.process();
+    } catch (Exception e) {
+      logger.error("Program Execution Error", e);
+    }
+  }
+
+  private void setInstrumentationKey() {
+    ApplicationInsightsAppender appender = (ApplicationInsightsAppender) logger.getAppender(
+      "aiAppender"
+    );
+    appender.setInstrumentationKey(env.get("APPINSIGHTS_INSTRUMENTATIONKEY"));
+  }
+
+  private synchronized void process() {
+    DefaultAzureCredential cred = new DefaultAzureCredentialBuilder().build();
+
+    QueueAsyncClient queueClient = new QueueClientBuilder()
+      .endpoint(env.get("AZURE_STORAGE_QUEUE_URI"))
+      .queueName(env.get("AZURE_STORAGE_QUEUE_NAME"))
+      .credential(cred)
+      .buildAsyncClient();
+
+    queueClient
+      .createWithResponse(Collections.singletonMap("queue", "metadataMap"))
+      .subscribe(
+        response -> logger.info("Queue Created: {}", response.getStatusCode()),
+        error -> System.err.print(error.toString())
+      );
+
+    while (true) {
```

Comment:
The more reactor way of doing the same would be:

```
     private void process() {
        DefaultAzureCredential credential = new DefaultAzureCredentialBuilder().build();
        QueueAsyncClient queueAsyncClient = new QueueClientBuilder()
            .endpoint("<endpoint>")
            .queueName("<queueName>")
            .credential(credential)
            .buildAsyncClient();

        queueAsyncClient.createWithResponse(Collections.singletonMap("queue", "metadataMap"))
            .doOnSuccess(response -> logger.info("Queue created: {}", response.getStatusCode()))
            .flatMapMany(response -> queueAsyncClient.receiveMessages(10).repeat())
            .flatMap(message -> {
                String msgId = message.getMessageId();
                logger.info("Message Received: {}, {}", msgId, message.getMessageText());
                return queueAsyncClient.deleteMessageWithResponse(msgId, message.getPopReceipt());
            }).subscribe(response -> logger.info("Deleted message: {}", response.getStatusCode()),
            error -> logger.error("Error: {}", error),
            () -> logger.trace("Receive message complete"));

        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            logger.error("Error while waiting", e);
        }
    }
```

Reply:
Very nice!  Thank you.

Reply:
This code wasn't working for me as-is. Would love to debug it with you sometime.  But for now, since I don't have much time. I'm going to go with my version...even though it's not as refactored as yours